plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveFileSystem.java

```diff
@@ -505,6 +506,120 @@ public void testFileIteratorListing()
+                .setTableName(table.getTableName())
+                .setDataColumns(ImmutableList.of(new Column("data", HIVE_LONG, Optional.empty())))
+                .setPartitionColumns(ImmutableList.of(new Column("part", HIVE_STRING, Optional.empty())))
+                .setOwner(Optional.empty())
+                .setTableType("fake");
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(StorageFormat.fromHiveStorageFormat(HiveStorageFormat.CSV));
+        Table fakeTable = tableBuilder.build();
+
+        // Expected file system tree:
+        // test-file-iterator-partitioned-listing/
+        //      .hidden/
+        //          nested-file-in-hidden.txt
+        //      part=simple/
+        //          _hidden-file.txt
+        //          plain-file.txt
+        //      part=nested/
+        //          parent/
+        //             _nested-hidden-file.txt
+        //             nested-file.txt
+        //      part=plus+sign/
+        //          plus-file.txt
+        //      part=level1|level2/
+        //          pipe-file.txt
+        //          parent1/
+        //             parent2/
+        //                nested-file.txt
+        //      part=level1 | level2/
+        //          pipe-blanks-file.txt
+        //      empty-directory/
+        //      .hidden-in-base.txt
+        Path basePath = new Path(getBasePath(), "test-file-iterator-partitioned-listing");
+        FileSystem fs = hdfsEnvironment.getFileSystem(TESTING_CONTEXT, basePath);
+        TrinoFileSystem trinoFileSystem = new HdfsFileSystemFactory(hdfsEnvironment, new TrinoHdfsFileSystemStats()).create(SESSION);
+        fs.mkdirs(basePath);
+
+        // create file in hidden folder
+        Path fileInHiddenParent = new Path(new Path(basePath, ".hidden"), "nested-file-in-hidden.txt");
+        fs.createNewFile(fileInHiddenParent);
+        // create hidden file in non-hidden folder
+        Path hiddenFileUnderPartitionSimple = new Path(new Path(basePath, "part=simple"), "_hidden-file.txt");
+        fs.createNewFile(hiddenFileUnderPartitionSimple);
+        // create file in `part=simple` non-hidden folder
+        Path plainFilePartitionSimple = new Path(new Path(basePath, "part=simple"), "plain-file.txt");
+        fs.createNewFile(plainFilePartitionSimple);
+        Path nestedFilePartitionNested = new Path(new Path(new Path(basePath, "part=nested"), "parent"), "nested-file.txt");
+        fs.createNewFile(nestedFilePartitionNested);
+        // create hidden file in non-hidden folder
+        Path nestedHiddenFilePartitionNested = new Path(new Path(new Path(basePath, "part=nested"), "parent"), "_nested-hidden-file.txt");
+        fs.createNewFile(nestedHiddenFilePartitionNested);
+        // create file in `part=plus+sign` non-hidden folder (which contains `+` special character)
+        Path plainFilePartitionPlusSign = new Path(new Path(basePath, "part=plus+sign"), "plus-file.txt");
```

Comment:
do we know that the path still contains the plus sign `+`?
would it be good to assert that, since that's the intention?

Reply:
i'll rather use native s3client to create the content to avoid any normalization happening through hive.

Reply:
i think we may need to have a "native" way to create test state on the storage.
with s3, this would go through s3client

in the absence of that, it still valluable to setup data with s3client (albeit limiting testing to s3)
this is exactly what i did in https://github.com/trinodb/trino/pull/18206